circuit/src/dynamic_sets/native.rs

```diff
@@ -868,23 +880,29 @@ mod test {
 			set.update_op(pks[i], op_i);
 		}
 
-		let (s, s_scaled) = set.converge();
+		let s = set.converge();
+		let s_ratios = set.converge_rational();
 
-		let s_formatted: Vec<String> = s.iter().map(|&x| fe_to_big(x).to_str_radix(10)).collect();
-		println!("new s: {:#?}", s_formatted);
-		let scaled_formatted: Vec<String> =
-			s_scaled.iter().map(|&x| fe_to_big(x).to_str_radix(10)).collect();
-		println!("scaled new s: {:#?}", scaled_formatted);
+		let mut tws = Vec::new();
+		let threshold = Fr::from_u128(435);
+		for (&score, ratio) in s.iter().zip(s_ratios.clone()) {
+			let tw = set.check_threshold::<NUM_LIMBS, POWER_OF_TEN>(score, ratio, threshold);
+			tws.push(tw);
+		}
+
+		(s, s_ratios, tws)
 	}
 
 	#[test]
 	fn test_scaling_1() {
 		const NUM_NEIGHBOURS: usize = 10;
 		const NUM_ITERATIONS: usize = 30;
 		const INITIAL_SCORE: u128 = 1000;
-		const SCALE: u128 = 100000000000000000000000000000000000000;
+		// Constants related to threshold check
+		const NUM_LIMBS: usize = 2;
+		const POWER_OF_TEN: usize = 50;
 
-		let ops = [
+		let ops_raw = [
```

Comment:
Here, the `ops` is simulated one, I think.
I mean the sum of one peer's opinion is the same among peers. (1000)
Have you tried any test of random opinion scores where the sum is different between peers?

Reply:
Yes, I tried, this just means we get much higher numbers for numerator and denominator as the result from `converge_rational`.
This would also mean that we have to use different params for `NUM_LIMBS` and `POWER_OF_TEN`. But to solve this, we will run the experiments with random values until we figure out what is the highest number of decimals we could get for numerator and denominator. Based on that we will scale our values and make some hardcoded params.
I will add this as TODO, since I need to focus on aggregators.